guide/src/migration.md

```diff
@@ -5,6 +5,59 @@ For a detailed list of all changes, see the [CHANGELOG](changelog.md).
 
 ## from 0.16.* to 0.17
 
+### Downcasting (`PyTryFrom`) has been changed for `PyMapping` and `PySequence` types
+
+Previously the safety checks for downcasting to `PyTryFrom` and `PyMapping`
+used the Python C-API functions `PyMapping_Check` and `PySequence_Check`.
+Unfortunately these functions are not sufficient for distinguishing such types,
+leading to inconsistent behavior (see
+[pyo3/pyo3#2072](https://github.com/PyO3/pyo3/issues/2072)).
+
+PyO3 0.17 changes these downcast checks to explicityly test if the type is a
+subclass of the corresponding abstract base class `collections.abc.Mapping` or
+`collections.abc.Sequence`. Note this requires calling into Python, which may
+incur a performance penalty over the previous method. If this performance
+penatly is a problem, you may be able to perform your own checks and use
+`try_from_unchecked` (unsafe).
+
+Another side-effect is that a pyclass defined in Rust with PyO3 will need to
+be _registered_ with the corresponding Python abstract base class for
+downcasting to succeed. `PySequence::register_abc_subclass` and
+`PyMapping:register_abc_subclass` have been added to make it easy to do this
```

Comment:
I wonder if they should just be named `PySequence::register` and `PyMapping::register`?

Reply:
Yeah I like that - it matches the Python function.